Advanced R book club - Chapter 2 document edits (#87)
r4ds/bookclub-advr: slides/02.qmd

## Learning objectives

> - To distinction between an *object* and its *name*
> - To identify when data are _physically_ copied versus modified
> - To better understand R's functional programming tools

Reviewer: ```suggestion
- Distinguish between an *object* and its *name*.
- Identify when data are *copied* versus *modified*.
- Trace the memory used by R.
```
I like to make the LOs more like sentences. I know the existing slides had "To...", so this wasn't your fault 😊

Data aren't really "physically" copied, so I changed the stress to "copied" vs "modified".

This chapter briefly mentions how this stuff is useful for functional programming, but we aren't really learning about those tools here, so I removed that LO, and added a more-specific one about what we're doing at the end of this chapter.

The lobstr package will help us through the chapter

```{r, echo = TRUE}
library(lobstr)
```

> ## Quiz

Reviewer: ```suggestion
## Quiz 1: Syntactic vs Non-syntactic names
```
No need for this to be a separate line on the slide.

> **1. Syntactic vs Non-syntactic names?**

Reviewer: ```suggestion
```

> Syntactic names: contain alphanumeric digits, `.`, and  `_`

Reviewer: ```suggestion
- **Syntactic names:** contain alphanumeric digits, `.`, and  `_`.
```

> Non-syntactic names: Start with a `_` or 0-9. Workaround is to surround in ```.

Reviewer: ```suggestion
- **Non-syntactic names:** 
  - Start with `_` or `0-9`. 
  - Workaround: surround in \`.
```

Also, can't use reserved words found at `?Reserved`. 

> ## Quiz

Reviewer: ```suggestion
## Quiz 2. On which line does `a` get copied
```

> **2. On which line does `a` get copied in the following example?**

Reviewer: ```suggestion
```

> ```{r, echo = TRUE}

Reviewer: ```suggestion
```{r}
```
Our defaults are TRUE for echo and eval (see `_metadata.yml`)

Author: For some reason it wasn't echo-ing when I previewed so I added them. But my mistake for not removing them! Will do in the update

Reviewer: It MIGHT depend how you preview. I'm guessing there's a way to render *just* this one file, and that probably won't pick up the site settings. Ctrl-shift-k on this file or `quarto render` on the site as a whole should render it properly.
